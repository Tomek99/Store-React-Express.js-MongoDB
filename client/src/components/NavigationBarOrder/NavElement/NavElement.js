import React from "react";
import styles from "./NavElement.module.scss";
import { AiOutlineCheckCircle } from "react-icons/ai";
import { Link } from "react-router-dom";

function NavElement({ item, id, tab }) {
  const { text, path } = item;
  const check = <AiOutlineCheckCircle size={25} color={`var(--text-color)`} />;

  const dot = (
    <span
      className={styles.dot}
      style={tab === id ? { color: "var(--main-color)" } : null}
    >
      {id}
    </span>
  );
  return (
    <div className={styles.NavElement}>
      <li className={styles.LiElement}>
        <Link
          to={path}
          className={
            tab > id ? styles.link : `${styles.link} ${styles.disabledLink}`
          }
        >
          <span style={id < tab ? null : { marginBottom: "3px" }}>
            {id < tab ? check : dot}
          </span>
          <span
            style={
              tab === id
                ? { color: "var(--main-color" }
                : { color: "var(--text-color)" }
            }
          >
            {text}
          </span>
        </Link>
      </li>
    </div>
  );
}

export default NavElement;
