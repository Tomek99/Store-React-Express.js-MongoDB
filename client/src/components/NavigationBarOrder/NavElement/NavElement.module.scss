.NavElement {
    width: 25%;
    position: relative;

    &::before {
        content: "";
        width: 70%;
        border-bottom: solid 2px rgba(255, 255, 255, 0.3);
        position: absolute;
        left: 65%;
        top: 12px;
        z-index: 1;
    }

    &:last-child::before {
        display: none;
    }
    .LiElement {
        display: flex;
        justify-content: center;
        align-items: center;

        .link {
            display: flex;
            flex-direction: column;
            align-items: center;
            text-align: center;
            font-size: 1.3rem;

            span {
                .dot {
                    display: flex;
                    justify-content: center;
                    align-items: center;
                    height: 25px;
                    width: 25px;
                    border: var(--border);
                    color: #ccc;
                    font-size: 1.4rem;
                    font-weight: bold;
                    border-radius: 50%;
                }
            }
        }

        .disabledLink {
            pointer-events: none;
        }
    }
}

@media (max-width: 500px) {
    .NavElement {
        &::before {
            width: 55% !important;
            left: 72.5% !important;
        }
    }
}
