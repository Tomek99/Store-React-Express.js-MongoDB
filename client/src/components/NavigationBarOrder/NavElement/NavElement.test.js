import { render } from "@testing-library/react";
import NavElement from "./NavElement";
import { MemoryRouter } from "react-router-dom";

const item = { text: "test", path: "/test" };
const id = "gsr21r21r";
const tab = 0;

test("should render NavElement component", () => {
  const { container } = render(
    <MemoryRouter>
      <NavElement item={item} id={id} tab={tab} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
