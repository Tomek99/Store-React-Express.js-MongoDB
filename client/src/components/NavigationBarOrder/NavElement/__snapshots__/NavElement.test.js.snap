// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavElement component 1`] = `
<div>
  <div
    class="NavElement"
  >
    <li
      class="LiElement"
    >
      <a
        class="link disabledLink"
        href="/test"
      >
        <span
          style="margin-bottom: 3px;"
        >
          <span
            class="dot"
          >
            gsr21r21r
          </span>
        </span>
        <span
          style="color: rgb(204, 204, 204);"
        >
          test
        </span>
      </a>
    </li>
  </div>
</div>
`;
